### PR TITLE
Added integration tests for vendor scenarios

### PR DESCRIPTION
**Pull request by felipe-saez**
Added integration tests to cover methods that return the new Vendor type.

### CODE REVIEW

File: example/integration_test/util/assertion_helper.dart [hunk 1]
```diff
@@ -3,11 +3,23 @@ import 'package:flutter/cupertino.dart';
...
+String extractTextFromWidget(String suffix) {
   final testKey = Key("nativeResponse_$suffix");
   final finder = find.byKey(testKey);
+  expect(finder, findsOneWidget);
   final text = finder.evaluate().single.widget as Text;
-  final actual = text.data!.removeNewLines();
-  assert(actual == expected, "Actual: $actual\nExpected: $expected");
-}
+  return text.data!.removeNewLines();
+}
```
Comment (nicolas-chaix-didomi) [inline example/integration_test/util/assertion_helper.dart:25]: please add an empty line at the end of the file

File: example/integration_test/vendor_test.dart [hunk 2]
```diff
@@ -259,15 +368,85 @@ void main() {
...
+    testWidgets("Get a vendor name with initialization and user agreed", (WidgetTester tester) async {
+      // Start app
+      app.main();
+      await tester.pumpAndSettle();
+
+      assert(isError == false);
+      assert(isReady == true);
+
+      await scrollDown(tester, listKey);
+      await scrollDown(tester, listKey);
```
Comment (nicolas-chaix-didomi) [inline example/integration_test/vendor_test.dart:444]: twice?

File: example/integration_test/util/assertion_helper.dart [hunk 3]
```diff
@@ -3,11 +3,23 @@ import 'package:flutter/cupertino.dart';
...
+
+/// Assert text in a Text widget whose key starts with "nativeResponse_" starts with the expected value being passed.
+Future<void> assertNativeMessageStartsWith(String suffix, String expected) async {
+  final actual = extractTextFromWidget(suffix);
+  assert(actual.startsWith(expected), "Actual: $actual\nExpected: $expected");
+}
+
+/// Return text from Text widget whose key starts with "nativeResponse_" and ends with suffix.
+String extractTextFromWidget(String suffix) {
```
Comment (nicolas-chaix-didomi) [inline example/integration_test/util/assertion_helper.dart:19]: extractTextFromWidget should be private or moved to a dedicated class?

File: example/integration_test/vendor_test.dart [hunk 4]
```diff
@@ -341,15 +506,85 @@ void main() {
...
+    testWidgets("Get a vendor name with initialization and user disagreed", (WidgetTester tester) async {
+      // Start app
+      app.main();
+      await tester.pumpAndSettle();
+
+      assert(isError == false);
+      assert(isReady == true);
+
+      await scrollDown(tester, listKey);
+      await scrollDown(tester, listKey);
```
Comment (nicolas-chaix-didomi) [inline example/integration_test/vendor_test.dart:582]: twice?

File: example/integration_test/vendor_test.dart [hunk 5]
```diff
@@ -177,15 +230,85 @@ void main() {
...
+    testWidgets("Get a vendor name with initialization", (WidgetTester tester) async {
+      // Start app
+      app.main();
+      await tester.pumpAndSettle();
+
+      assert(isError == false);
+      assert(isReady == true);
+
+      await scrollDown(tester, listKey);
+      await scrollDown(tester, listKey);
```
Comment (nicolas-chaix-didomi) [inline example/integration_test/vendor_test.dart:306]: twice?

File: example/integration_test/vendor_test.dart [hunk 6]
```diff
@@ -96,15 +93,85 @@ void main() {
...
+
+    testWidgets("Get a vendor name without initialization", (WidgetTester tester) async {
+      // Start app
+      app.main();
+      await tester.pumpAndSettle();
+
+      assert(isError == false);
+      assert(isReady == false);
+
+      await scrollDown(tester, listKey);
```
Comment (nicolas-chaix-didomi) [inline example/integration_test/vendor_test.dart:168]: twice?

File: example/integration_test/util/assertion_helper.dart [hunk 7]
```diff
@@ -3,11 +3,22 @@ import 'package:flutter/cupertino.dart';
...
   final testKey = Key("nativeResponse_$suffix");
   final finder = find.byKey(testKey);
+  expect(finder, findsOneWidget);
   final text = finder.evaluate().single.widget as Text;
-  final actual = text.data!.removeNewLines();
-  assert(actual == expected, "Actual: $actual\nExpected: $expected");
-}
+  return text.data!.removeNewLines();
+}
```
Comment (pmerlet-at-didomi) [inline example/integration_test/util/assertion_helper.dart:25]: Missing blank line

Comment (felipe-saez) [reply example/integration_test/util/assertion_helper.dart]: done

File: example/integration_test/util/assertion_helper.dart (hunk 1 above)
Comment (felipe-saez) [reply example/integration_test/util/assertion_helper.dart]: done

File: example/integration_test/util/assertion_helper.dart (hunk 3 above)
Comment (felipe-saez) [reply example/integration_test/util/assertion_helper.dart]: I ended up making the method private

**Review by nicolas-chaix-didomi — APPROVED** [review]

**Pull request merged**